### ISSUE TITLE
"Client is trying to use protocol=2 but server is set to protocol=2"

### ISSUE THREAD

**Issue by sberthelot** [grke/burp#513]
When restoring for an alternate client (-C), if this client has a "protocol=X" set, we get an error message like the one in the subject.

I had to specify protocol 2 on the server for all clients since the default has changed to protocol 1 some versions ago.

On `src/server/extra_comms.c:336` there is `if(protocol!=PROTO_AUTO)`, shouldn't the comparison be between `protocol` and `cliproto` (with atoi or similar) instead ?

**Comment by grke** [comment]
It sounds like you are using 2.0.50, not 2.0.52, because line 336 no longer looks like that. Maybe you mean line 284.

If the server is deciding what protocol all the clients should use, you should set your clients to be protocol=0 (or not set it). The idea is that only one side can decide.
But, I see your point in that if both sides want to force the same protocol, that should probably be allowed.
I don't think it is worth fixing this for the stable release though.

**Comment by sberthelot** [comment]
I just checked and the server producing this message is using 2.0.52, I was trying to restore a client on the server itself so both are using the same version.

I initially had this message with proto 1 and proto 2 so it seemed legit to me but seing it also with same proto is rather strange.

I have looked at git master , to be sure here is the URL : https://github.com/grke/burp/blob/master/src/server/extra_comms.c
The check is on line 336 for me or maybe I missed something.

I remember we had to set the proto on clients before (maybe a bug or strange behavior on old version I don't remember why I just remember we had to do it), we can remove it there and set it on servers to see if everything is OK now (also with burp 1 servers).

Meanwhile even if it's not a blocker bug this message is rather strange and would be better to me if it  either explains than both cannot set protocol (any version even identical) or let them go through if they are the same...
